src/OpenLoco/Windows/TerraForm.cpp

```diff
@@ -329,7 +328,7 @@ namespace OpenLoco::Ui::Windows::Terraform
             if (!Input::hasFlag(Input::Flags::toolActive))
                 WindowManager::close(self);
 
-            if (_toolWindowType != WindowType::terraform)
+            if (WindowManager::getToolWindowType() != WindowType::terraform)
```

Comment:
Again can use `if (!isToolActive(WindowType::terraform))` and skip the duplicated close. Also after a close it should return from this function.